src/lsptoolshost/nestedCodeAction.ts

```diff
@@ -0,0 +1,131 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { CodeAction, CodeActionResolveRequest, LSPAny } from 'vscode-languageserver-protocol';
+import { RoslynLanguageServer } from './roslynLanguageServer';
+import { URIConverter, createConverter } from 'vscode-languageclient/lib/common/protocolConverter';
+import { UriConverter } from './uriConverter';
+import { getFixAllResponse } from './fixAllCodeAction';
+
+export function registerNestedCodeActionCommands(
+    context: vscode.ExtensionContext,
+    languageServer: RoslynLanguageServer,
+    outputChannel: vscode.OutputChannel
+) {
+    context.subscriptions.push(
+        vscode.commands.registerCommand(
+            'roslyn.client.nestedCodeAction',
+            async (request): Promise<void> => registerNestedResolveCodeAction(languageServer, request, outputChannel)
+        )
+    );
+}
+
+async function registerNestedResolveCodeAction(
+    languageServer: RoslynLanguageServer,
+    codeActionData: any,
+    outputChannel: vscode.OutputChannel
+): Promise<void> {
+    if (codeActionData) {
+        const data = <LSPAny>codeActionData;
+        const action = data.NestedCodeAction;
+
+        if (action?.nestedActions?.length > 0) {
+            const codeActionTitles: string[] = getCodeActionTitles(action.nestedActions);
+            const selectedValue = await vscode.window.showQuickPick(codeActionTitles, {
+                placeHolder: vscode.l10n.t(action.title),
+                ignoreFocusOut: true,
+            });
+            if (selectedValue) {
+                const selectedAction = retrieveSelectedAction(selectedValue, action.nestedActions);
```

Comment:
I think we can make this much easier and entirely delete `retrieveSelectedAction`

Take a look at https://github.com/dotnet/vscode-csharp/blob/6e35fe33f77060f651e81c27450479fc847a0156/src/shared/assets.ts#L64

Basically - where you pass in `codeActionTitles` to `showQuickPick`, you would instead pass in an object that contains a `label` field and a field with the code action object itself.  Then the `selectedAction` returned will have the actual code action object already, so you don't need to search for it again.